runtime_lib/controller/main.cpp

```diff
@@ -897,7 +897,7 @@ void unlock_uart() {
 int queue_create(uint32_t size, queue_t **queue, uint32_t mb_id) {
   uint64_t queue_address[1] = {base_address + sizeof(dispatch_packet_t)};
   uint64_t queue_base_address[1] = {queue_address[0] +
-                                    sizeof(dispatch_packet_t)};
+                                    2 * sizeof(dispatch_packet_t)};
```

Comment:
This seems like an important fix separate from the platform version. I think we are skipping over the queue data structure here. To avoid future bugs maybe it should be something like:
```C++
ceil((float)sizeof(queue_t) / (float)sizeof(dispatch_packet_t)) * sizeof(dispatch_packet_t)
```
note `queue_base_address` must be aligned to the packet size per hsa spec.

Reply:
what about something like this? It avoids using floating point, at the cost of the loop, but the number of iteration should be small

```
uintptr_t queue_base_address_min = queue_address[0] + sizeof(queue_t);
uintptr_t queue_base_address[1] = {queue_address[0] + sizeof(dispatch_packet_t);}
while (queue_base_address[0] < queue_base_address_min) {
  queue_base_address[0] += sizeof(dispatch_packet_t);
}
```